internal/simplecue/generator.go

```diff
@@ -113,12 +113,9 @@ func (g *generator) walkCueSchemaWithEnvelope(envelopeName string, v cue.Value)
 	for i.Next() {
 		if i.Selector().IsDefinition() {
 			name := g.namingFunc(g.rootVal, i.Value().Path())
-			n, err := g.declareObject(name, i.Value())
-			if err != nil {
+			if err := g.declareObject(name, i.Value()); err != nil {
 				return err
 			}
-
-			g.schema.AddObject(n)
```

Comment:
I don't `g.schema.AddObject(n)` called anywhere else. Was this removed intentionally?

Reply:
It's done directly in `declareObject()` now.

I found it a bit misleading to have a "declareObject" function that didn't actually add the object to the schema :sweat_smile:

Reply:
Ah okay! I did see that called, but wasn't sure if it was the equivalent. Agreed this is much clearer to have it done once, then in each iteration.